contracts/standard/arbitration/MultipleArbitrableTransaction.sol

```diff
@@ -0,0 +1,320 @@
+        Transaction storage transaction = transactions[_transactionId];
+        require(msg.sender == transaction.buyer || msg.sender == transaction.seller);
+        
+        require(transaction.status>=Status.DisputeCreated);
+        Evidence(_transactionId, transaction.arbitrator,transaction.disputeId,msg.sender,_evidence);
+    }
+    
+    /** @dev Appeal an appealable ruling.
+     *  Transfer the funds to the arbitrator.
+     *  Note that no checks are required as the checks are done by the arbitrator.
+     *  @param _transactionId The index of the transaction.
+     *  @param _extraData Extra data for the arbitrator appeal procedure.
+     */
+    function appeal(uint _transactionId, bytes _extraData) payable public {
+        Transaction storage transaction = transactions[_transactionId];
+        require(msg.sender == transaction.buyer || msg.sender == transaction.seller);
+        
+        transaction.arbitrator.appeal.value(msg.value)(transaction.disputeId,_extraData);
+    }
+    
+
+
+    /** @dev
+     *  @param _arbitrator The arbitrator of the contract.
+     *  @param _hashContract Keccak hash of the plain English contract.
+     *  @param _timeout Time after which a party automatically loose a dispute.
+     *  @param _seller The recipient of the transaction.
+     *  @param _arbitratorExtraData Extra data for the arbitrator.
+     */
+    function createTransaction(Arbitrator _arbitrator, bytes32 _hashContract, uint _timeout, address _seller, bytes _arbitratorExtraData) payable public returns (uint transactionIndex) {
+        transactions.push(Transaction({ 
+            seller: _seller,
+            buyer: msg.sender,
+            amount: msg.value,
+            timeout: _timeout,
+            disputed: false,
+            arbitrator: _arbitrator,
+            arbitratorExtraData: _arbitratorExtraData,
+            disputeId: 0,
+            sellerFee: 0,
+            buyerFee: 0,
+            lastInteraction: now,
+            status: Status.NoDispute
+        }));
+        ContractHash(transactions.length - 1, _hashContract);
+        return transactions.length - 1;
+    }
+
+    /** @dev
+     * returns the last transaction id whose buyer is the sender
+     */
+    function lastTransactionId () constant public returns (uint index) {
+        uint len = transactions.length;
+        for (uint i = len - 1; i>=0; i--) {
+            if (transactions[i].buyer == msg.sender) {
+                return i;
+            }
+        }
+    }
+
+    /** @dev
+     * @return amount The transaction amount. Can be called by buyer, seller or arbitrator.
+     * @param _transactionId The index of the transaction.
+     */
+    function amount (uint _transactionId) constant public returns (uint amount) {
+        Transaction storage transaction = transactions[_transactionId];
+        require(msg.sender == transaction.buyer || msg.sender == transaction.seller || msg.sender == address(transaction.arbitrator));
```

Comment:
This access control is not required because it's a constant function.
Note that everyone can read smart contracts variable so access control can only prevent state modification but never state read.